cli/tools/bundle/mod.rs

```diff
@@ -0,0 +1,879 @@
+use std::path::Path;
+use std::path::PathBuf;
+use std::rc::Rc;
+use std::sync::Arc;
+
+use deno_ast::ModuleSpecifier;
+use deno_cache_dir::GlobalOrLocalHttpCache;
+use deno_config::deno_json::TsTypeLib;
+use deno_core::anyhow;
+use deno_core::error::AnyError;
+use deno_core::resolve_url_or_path;
+use deno_core::url::Url;
+use deno_core::ModuleLoader;
+use deno_error::JsError;
+use deno_graph::Position;
+use deno_lib::worker::ModuleLoaderFactory;
+use deno_npm::npm_rc::ResolvedNpmRc;
+use deno_npm::registry::NpmRegistryApi;
+use deno_npm_cache::TarballCache;
+use deno_resolver::npm::managed::ResolvePkgFolderFromDenoModuleError;
+use deno_resolver::workspace::WorkspaceNpmPatchPackages;
+use deno_runtime::deno_permissions::PermissionsContainer;
+use deno_semver::npm::NpmPackageReqReference;
+use deno_semver::package::PackageNv;
+use esbuild_rs::protocol;
+use esbuild_rs::EsbuildFlagsBuilder;
+use esbuild_rs::EsbuildService;
+use indexmap::IndexMap;
+use node_resolver::errors::PackageSubpathResolveError;
+use node_resolver::NodeResolutionKind;
+use node_resolver::ResolutionMode;
+use sys_traits::EnvCurrentDir;
+
+use crate::args::BundleFlags;
+use crate::args::BundleFormat;
+use crate::args::Flags;
+use crate::args::PackageHandling;
+use crate::cache::DenoDir;
+use crate::factory::CliFactory;
+use crate::graph_container::MainModuleGraphContainer;
+use crate::graph_container::ModuleGraphContainer;
+use crate::graph_container::ModuleGraphUpdatePermit;
+use crate::module_loader::ModuleLoadPreparer;
+use crate::module_loader::PrepareModuleLoadOptions;
+use crate::node::CliNodeResolver;
+use crate::npm::CliNpmCache;
+use crate::npm::CliNpmCacheHttpClient;
+use crate::npm::CliNpmRegistryInfoProvider;
+use crate::npm::CliNpmResolver;
+use crate::resolver::CliNpmReqResolver;
+use crate::resolver::CliResolver;
+use crate::sys::CliSys;
+
+const ESBUILD_VERSION: &str = "0.25.5";
+
+pub async fn ensure_esbuild(
```

Comment:
Since this is an implementation detail, maybe you could move it to `bundle/esbuild.rs` module?